### PR TITLE
BAU: Add data directory version check

### LINKED ISSUE #11: Feature Request: When `./data` is old and out of date apps fail to start
Maybe delete it and regenerate if it's older than a week or two?

### PR DESCRIPTION
**Pull request by andy-paine**
- Assume that if the generate directory files changed, the
PKI needs to be updated
- shasum all files in generate directory and output to .version

Fixes #11

### CODE REVIEW

File: startup.sh [hunk 1]
```diff
@@ -12,7 +13,8 @@ EOF
 tput sgr0
 
 # Generate PKI and config if necessary
-if test ! -d data; then
+data_version="$(./generate/get-version.sh)"
```
Comment (rhowe-gds) [inline startup.sh]: If you're relying on relative paths being meaningful, you probably want to do this at the start of your script

`cd "$(dirname "$0")"`

File: generate/get-version.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+script_dir="$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)"
+find $script_dir -type f | xargs cat | shasum | cut -d ' ' -f 1
```
Comment (rhowe-gds) [inline generate/get-version.sh]: Why not just `find "$(dirname "${BASH_SOURCE[0]}")" ....`?

File: startup.sh [hunk 3]
```diff
@@ -12,7 +13,8 @@ EOF
...
+if test ! -d data -o ! "$(cat data/.version 2>/dev/null)" == $data_version; then
```
Comment (rhowe-gds) [inline startup.sh]: `$(cat foo)` is `$(< foo)` in bash

Comment (rhowe-gds) [reply startup.sh]: However, check out the `-c` option to `shasum`

File: generate/get-version.sh (hunk 2 above)
Comment (rhowe-gds) [reply generate/get-version.sh]: Why are you shasuming the cat'd contents of all the scripts?
If that's really what you wanted to do you may as well

`tar cp "$script_dir" | shasum | cut -d\  -f 1`

**Comment by rhowe-gds** [comment]
You could also just use make, but then you're on a slippery slope

**Comment by andy-paine** [comment]
Nice idea, badly implemented by me. I might revisit this later.

**Pull request closed without merging**